packages/SwingSet/src/kernel/dynamicVat.js

```diff
@@ -13,34 +14,56 @@ export function makeDynamicVatCreator(stuff) {
     addVatManager,
     addExport,
     queueToExport,
+    getBundle,
+    kernelKeeper,
+    panic,
   } = stuff;
 
   /** A function to be called from the vatAdmin device to create a new vat. It
    * creates the vat and sends a notification to the device. The root object
    * will be available soon, but we immediately return the vatID so the ultimate
    * requestor doesn't have to wait.
    *
-   * @param vatSourceBundle a source bundle (JSON-serializable data) which
-   * defines the vat. This should be generated by calling bundle-source on a
-   * module whose default export is makeRootObject(), which takes E as a
-   * parameter and returns a root object.
-   * @param options an options bundle. The only option defined so far is
-   * 'metered', which defaults to 'true'. If 'true', the new dynamic vat is
-   * subject to a meter that limits the amount of computation and allocation
-   * that can occur during any given crank. Stack frames are limited as well.
-   * The meter is refilled between cranks, but if the meter ever underflows,
-   * the vat is terminated. If 'false', the vat is unmetered.
+   * @param source an object which either has a `bundle` (JSON-serializable
```

Comment:
_Really_ not a fan of switching from two methods with simple APIs to one method with a complicated API.

Reply:
I hear you, but we can't simplify it to the point that `dynamicVat.js` only hears about full bundle objects, as it did before. If we did that, the create-from-name case would throw away the name before `dynamicVat.js` heard about it, which means it would have to store the full (unnamed) bundle into the DB for both cases, not just create-from-bundle. By retaining the name (when one is available), we can store just the name in the DB. And that feels like it's going to be important for storage-performance pretty quickly.

The fully-denormalized(?) alternative would probably be for `vatKeeper` to have two separate methods (one that returns a source bundle or `undefined`, a second which returns a source bundleName or undefined). The kernel would call both in succession to see which one has a value. Then `dynamicVat.js` must expose *four* methods (first-time-vs-replay times name-vs-bundle). That felt like a net complexity increase over the polymorphic `source` argument. We could have it expose two methods (name-vs-bundle) and make both take an optional `vatID` to select between first-time-vs-replay, but I was more uncomfortable using polymorphism over `vatID` than I was over `source`.